From 83ecafcda5326baceeb68fba90dc13083ff48fe6 Mon Sep 17 00:00:00 2001
From: Zhipeng Xie <xiezhipeng1@huawei.com>
Date: Fri, 2 Nov 2018 17:25:50 +0000
Subject: [PATCH 1014/1015] create-diff-object: add dynamic reloction for
 function pointer on aarch64

implement function_ptr_rela for aarch64.

Signed-off-by: Zhipeng Xie <xiezhipeng1@huawei.com>
---
 kpatch-build/create-diff-object.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/kpatch-build/create-diff-object.c b/kpatch-build/create-diff-object.c
index e72f6dd..f8f3e15 100644
--- a/kpatch-build/create-diff-object.c
+++ b/kpatch-build/create-diff-object.c
@@ -2628,7 +2628,9 @@ static int function_ptr_rela(const struct rela *rela)
 		rela_toc->addend == (int)rela_toc->sym->sym.st_value &&
 		(rela->type == R_X86_64_32S ||
 		rela->type == R_PPC64_TOC16_HA ||
-		rela->type == R_PPC64_TOC16_LO_DS));
+		rela->type == R_PPC64_TOC16_LO_DS ||
+		rela->type == R_AARCH64_ADR_PREL_PG_HI21 ||
+		rela->type == R_AARCH64_ADD_ABS_LO12_NC));
 }
 
 static int may_need_dynrela(const struct rela *rela)
-- 
1.7.5.4

